security/pkg/pki/ca/ca_test.go

```diff
@@ -93,62 +96,101 @@ func TestCreateSelfSignedIstioCAWithoutSecret(t *testing.T) {
 	maxCertTTL := time.Hour
 	org := "test.ca.Org"
 	const caNamespace = "default"
-	client := fake.NewSimpleClientset()
 	rootCertFile := ""
 	rootCertCheckInverval := time.Hour
-	rsaKeySize := 2048
 
-	caopts, err := NewSelfSignedIstioCAOptions(context.Background(),
-		0, caCertTTL, rootCertCheckInverval, defaultCertTTL,
-		maxCertTTL, org, false, caNamespace, client.CoreV1(),
-		rootCertFile, false, rsaKeySize)
-	if err != nil {
-		t.Fatalf("Failed to create a self-signed CA Options: %v", err)
+	cases := []struct {
+		name          string
+		algorithmType string
+		eccSigAlg     string
+		eccCurve      string
+		rsaKeySize    int
+	}{
+		{
+			name:          "Test RSA",
+			algorithmType: "RSA",
+			rsaKeySize:    2048,
+		},
+		{
+			name:          "Test ECC",
+			algorithmType: "ECC",
+			eccSigAlg:     "ECDSA",
+			eccCurve:      "P256",
+		},
```

Comment:
nit: add additional test case for an invalid eccSigAlg/eccCurve

Reply:
added more test cases